sonic-xcvrd/xcvrd/xcvrd_utilities/media_settings_parser.py

```diff
@@ -48,24 +52,42 @@ def media_settings_present():
 
 
 def get_lane_speed_key(physical_port, port_speed, lane_count):
+    """
+    Get lane speed key for the given port
+
+    Args:
+        physical_port: physical port number for this logical port
+        port_speed: logical port speed in Mbps
+        lane_count: number of lanes for this logical port
+
+    Returns:
+        the lane speed key string, in either host electrical interface format (e.g: 'speed:200GAUI-8')
+        or regular format (e.g: 'speed:25G')
+        Refer to Table 4-5 of SFF-8024 for different kinds of host electrical interfaces.
+    """
     sfp = xcvrd.platform_chassis.get_sfp(physical_port)
     api = sfp.get_xcvr_api()
     
     lane_speed_key = None
     if xcvrd.is_cmis_api(api):
         appl_adv_dict = api.get_application_advertisement()
-        app_id = xcvrd.get_cmis_application_desired(api, int(lane_count), int(port_speed))
+        app_id = xcvrd.get_cmis_application_desired(api, lane_count, port_speed)
         if app_id and app_id in appl_adv_dict:
             host_electrical_interface_id = appl_adv_dict[app_id].get('host_electrical_interface_id')
             if host_electrical_interface_id:
                 lane_speed_key = LANE_SPEED_KEY_PREFIX + host_electrical_interface_id.split()[0]
-
+    if not lane_speed_key:
```

Comment:
@longhuan-cisco would suggest to use else of `if xcvrd.is_cmis_api(api)` so that we know this is a non-CMIS case